README.md

```diff
@@ -7,3 +7,258 @@ instances for parsing and rendering. A convenient DSL to define those
+a type `Status` that can have only values `Active`, `Pending` or `Deleted`:
+
+```haskell
+"Status" =:= enum Both
+  & "Active"
+  & "Pending"
+  & "Deleted"
+```
+
+This definition will generate the following type for us:
+
+```haskell
+data XmlStatus
+  = XmlStatusActive
+  | XmlStatusPending
+  | XmlStatusDeleted
+  | UnknownXmlStatus !String
+```
+
+It has all the necessary instances, so you can use it as a type for a field.
+
+### Append content
+
+Sometimes XML you are dealing with contains a mix of elements and immediate content.
+Something like the following:
+
+```XML
+<?xml version="1.0" encoding="UTF-8"?>
+<example>
+    <field1>
+        I am
+    </field1>
+    totally
+    <field2>
+        weird
+    </field2>
+    LOL
+</example>
+```
+
+You can model this with an "append content" modifier `^`. It will instruct
+`xml-isogen` to append content of the field as it is, without wrapping it
+into an XML element. For our case it may look like this:
+
+```haskell
+"Example2" =:= record Both
+  ! "field1" [t|Text|]
+  ^ "mixed1" [t|Text|]
+  ! "field2" [t|Text|]
+  ^ "mixed2" [t|Text|]
+```
+
+### Attributes
+
+`xml-isogen` also supports XML attributes using `!%` and `?%` modifiers:
+
+```haskell
+"Example3" =:= record Both
+  ! "field1" [t|Text|]
+  !% "attribute1" [t|Text|]
+  ?% "attribute2" [t|Text|]
+
+"Body" =:= record Both
+  ! "root" [t|XmlExample3|]
+```
+
+The following two types will be generated:
+
+```haskell
+data XmlExample3
+  = XmlExample3 {_xe3Field1 :: !Text,
+                 _xe3Attribute1 :: !Text,
+                 _xe3Attribute2 :: !(Maybe Text)}
+
+newtype XmlBody = XmlBody {_xbRoot :: XmlExample3}
+```
+
+The `_xe3Attribute2` is optional because we used `?%` modifier. Attributes will be
+attached to parent XML element. Here is an example of the generated XML file:
+
+```XML
+<root attribute1="world">
+    <field1>
+	hello
+    </field1>
+</root>
+```
+
+Note that attributes are attached to the parent XML element, that's why we needed
+`XmlBody` type here.
+
+### Namespaces
+
+Often XSD schema requires XML elements to be qualified with a namespace. To instruct
+`xml-isogen` to qualify fields, specify namespace is a curly brackets:
+
+```haskell
+"Example4" =:= record Both
+  ! "field1" [t|Text|]
+  ! "{http://example.com/1}field2" [t|Text|]
+  ! "{http://example.com/2}field3" [t|Text|]
+```
+
+Here is the generated XML:
+
+```XML
+<field1>
+    hello
+</field1>
+<field2 xmlns="http://example.com/1">
+    world
+</field2>
+<field3 xmlns="http://example.com/2">
+    !
+</field3>
+```
+
+### Nillable types
+
+Sometimes optional element in XML are encoded using `nill="true"` attribute instead of
+omitting the element. (The `nill` attribute comes from `http://www.w3.org/2001/XMLSchema-instance` namespace). With `xml-isogen` you handle it using `Nillable` type:
+
+```haskell
+"Example5" =:= record Both
+  ! "field" [t|Nillable Text|]
+```
+
+If the field has value `Nothing`, then the following XML will be generated:
```

Comment:
field ~~has~~ **contains the** value `Nothing`,

Reply:
And it would probably be better to explicitly show the record, from which the xml is generated